lib/galaxy/managers/collections.py

```diff
@@ -215,10 +215,20 @@ def collection_builder_for(self, dataset_collection):
         collection_type_description = self.collection_type_descriptions.for_collection_type(collection_type)
         return builder.BoundCollectionBuilder(dataset_collection, collection_type_description)
 
-    def delete(self, trans, instance_type, id):
+    def delete(self, trans, instance_type, id, recursive=False, purge=False):
```

Comment:
Would this delete datasets that exist outside of collections or that exist in copied collections ?

Reply:
Yes. We duplicate the HDAs when we copy HDCAs though (that is what I claimed when I investigated it here https://github.com/galaxyproject/galaxy/pull/5004 anyway - if that isn't the case we can fix that). We also duplicate the HDAs when copying them for collection operations. So I'd like to get to the point where having an HDA belong to multiple collections be something that is supported on the backend but not ever actually used in in the GUI. We are also taking steps to eliminate the need to unhide datasets in order to use them #5384 - so hopefully that will happen less - but if that is happening presumably the researcher understands that HDA was part of the collection.

I can add some mouseover text to the various deletion options in the GUI to go into greater details though if there is a concern users might accidentally delete files they don't intend to. Is that enough?

Reply:
Ah alright, that is good enough then, no need for additional text!